matter_server/server/device_controller.py

```diff
@@ -949,11 +950,10 @@ def attribute_updated_callback(
                 attribute_updated, path, old_value, new_value
             )
 
-        def event_callback(
+        def event(
```

Comment:
Where is this called ?

Reply:
Looking at the code this seems to be dead code as the logic is handled in "event_callback"

Reply:
This is called via `self._loop.call_soon_threadsafe`. The diff is a bit hard to read, but essentially everything in `event_callback` was moved to the function just named `event`.

Reply:
OK, in that case maybe the function needs to be renamed to "on_event" or something like that to indicate its still a callback ?

Reply:
Agreed, just `event` is really confusing. But it did align well with what we had with `attribute_updated_callback` (and `attribute_updated`).

Essentially, what we do here is we convert a threaded callback into a callback on the event loop. So I came up with this now:

- `attribute_updated_callback_threadsafe` calls `attribute_updated_callback` on the event loop
- `event_callback_threadsafe` calls `event_callback` on the event loop

Reply:
Btw, I did check the code, these are the only two callbacks (from the subscription callbacks) which are called straight from the CHIP SDK thread currently.

Reply:
Anything called on the event loop should be prefixed with `async_`

Reply:
Hm, I'd say that should essentially be pretty much everything in this case no :thinking: . Does that make sense for a project which started off with asyncio?

Reply:
> Anything called on the event loop should be prefixed with `async_`

In HA core I agree (as there is still a large part non-async) but our code base is started with asyncio in mind and we are talking about a handful of non-async calls so if imo prefixing with async_ is a bit meh. I'd rather prefix the non-async friendly ones or mention in the docstring that its not async friendly.

Reply:
> Anything called on the event loop should be prefixed with `async_`

Ok, we kinda went the other way here, and essentially marked thread safe functions now.

I think it is already better what we have, so I'll move forward with this as is. IMHO, if we decide to prefix everything with `async_` would be a bigger change and should be a separate PR.